source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,237 @@
+===========
+Aggregation
+===========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Aggregation operations process data in your MongoDB collections and return computed results. The **aggregation pipeline**
+is like a factory assembly line. Along the line are assembly stations, called **aggregation stages** . Like an assembly
+station configured with specific tooling to perform its task, each ``stage`` uses **expressions** to perform some
+operation or combination of operations.
+
+Aggregation Does More
+~~~~~~~~~~~~~~~~~~~~~
+
+Using ``find`` operations, you can:
+
+- select *what* documents to return
+- select *what* fields to return
+- sort the results
+
+Using ``aggregation`` operations, you can:
+
+- perform all ``find`` operations
+- rename fields
+- calculate fields
+- summarize data
+- group values
+
+Illustrated Example
+~~~~~~~~~~~~~~~~~~~
+
+In a car factory assembly line, one station might shape and bend the frame using a hydraulic press and a welder, another
+might affix the doors with a drill, and so on.
+
+In the following animation, every document in the collection enters the pipeline and is first filtered by
+the ``$match`` stage. In this example, the stage selectively matches documents that are either "red" or "blue",
+filtering "green" documents out. The documents continue flowing through the pipeline and enter the ``$project`` stage,
+which transforms them from squares to circles. Lastly, the documents flow into the ``$group`` stage, configured to
+produce the ratio of red to blue circles.
+
+.. figure:: /figures/agg_factory.gif
+   :border:
+   :alt: Animation showing how the aggregation pipeline works, conceptually.
+
+
+Using the **Aggregation builders** API, the same operation may look something like:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/agg-factory.java
+   :language: java
+
+.. code-block:: json
+
+   {"_id": 0, "red": 3, "blue": 2}
+
+Terms
+~~~~~
+
+- :manual:`Aggregation pipeline </core/aggregation-pipeline/>`
+- :manual:`Aggregation stages </meta/aggregation-quick-reference/#stages>`
+- :manual:`Operator expressions </meta/aggregation-quick-reference/#operator-expressions>`
+- :ref:`Aggregation Builders <builders>`
+
+Runnable Examples
```

Comment:
Would you consider adding sub-headings under this for "Setup" and then "Aggregation Example"? When I read this heading, I expect to see the sample code rather than the initial setup so it could be helpful to identify what's happening in each paragraph.